manifests/init.pp

```diff
@@ -75,6 +77,35 @@
     $system_environment != 'docker'
   )
 
+  # Defaults for specific platforms
+  case $::osfamily {
+    'Debian','Suse': {
+      $def_umask = '027'
+      $def_sys_uid_min = 100
+      $def_sys_gid_min = 100
+      $shadowgroup = 'shadow'
+      $shadowmode = '0640'
+    }
+    'RedHat': {
+      $def_umask = '077'
+      $def_sys_uid_min = 201
+      $def_sys_gid_min = 201
+      $shadowgroup = 'root'
+      $shadowmode = '0000'
+    }
+    default: {
+      $def_umask = '027'
+      $def_sys_uid_min = 100
+      $def_sys_gid_min = 100
+      $shadowgroup = 'root'
+      $shadowmode = '0600'
+    }
+  }
+
+  # Merge defaults
+  $merged_umask = pick($umask, $def_umask)
```

Comment:
@mcgege is the `pick` function something new in puppet? Could you maybe link me to the docs (I can't find them on the quick view)?

Reply:
@artem-sidorenko It's a function of stdlib: [pick](https://github.com/puppetlabs/puppetlabs-stdlib#prefix)